test/Tests/src/Semantic/Python_Interop_Spec.enso

```diff
@@ -0,0 +1,95 @@
+from Base import all
+import Test
+
+foreign python my_method a b = """
+    return a + b
+
+type My_Type
+    type My_Type a b
+
+    foreign python my_method = """
+        return self.a + self.b
+
+    my_method_2 x = this.my_method * x
+
+    foreign python my_method_3 y = """
+        r = self.my_method_2(y)
+        return r + 1
+
+    foreign python my_throw = """
+        err = RuntimeError('Error!')
+        raise err
+
+    do_throw = Panic.throw this
+
+    foreign python do_catch = """
+        try:
+            self.do_throw()
+        except Exception as e:
+            return e.a
+
+foreign python make_object = """
+    class My:
+        def __init__(self):
+            self.x = 10
+            self.y = False
+        def compare(self, guess):
+            return self.x < guess
+    return My()
+
+foreign python make_array = """
+    class My:
+        def __init__(self, x):
+            self.x = x
+        def compare(self, guess):
+            return self.x < guess
+    return [My(10), My(20), My(30)]
+
+foreign python make_str str = """
+    return ("foo " + str + " bar")
+
+spec =
+    pending = if Polyglot.is_language_installed "python" then Nothing else """
+        Can't run Python tests, Python is not installed.
+    Test.group "Polyglot Python" pending=pending <|
+        Test.specify "should allow declaring module-level methods in JS" <|
+            here.my_method 1 2 . should_equal 3
+
+        Test.specify "should allow mutual calling of instance-level methods" <|
+            My_Type 3 4 . my_method_3 5 . should_equal 36
+
+        Test.specify "should expose methods and fields of JS objects" <|
+            obj = here.make_object
+            obj.x . should_equal 10
+            obj.y . should_be_false
+            obj.compare 5 . should_be_false
+            obj.compare 11 . should_be_true
+
+        Test.specify "should expose array interfaces for JS arrays" <|
+            vec = Vector.Vector here.make_array
+            vec.map .x . should_equal [10, 20, 30]
+
+        Test.specify "should correctly marshall strings" <|
+            str = here.make_str "x" + " baz"
+            str.should_equal "foo x bar baz"
+
+        Test.specify "should make Python strings type pattern-matchable" <|
+            str = here.make_str "x"
+            t = case str of
+                Text -> True
+                _ -> False
+            t.should_be_true
```

Comment:
We should probably have a similar test for "make python numbers type pattern-matchable". Same for JS if we don't already.